### DISCUSSION TITLE
Github action  env reference to azure terraform variables

### DISCUSSION THREAD

**Discussion by nehabam** [community/community#30042] [Actions]
I am trying to use github action for terraform multistage deployment.  i want to know how to create a custom variable in github action level which will refer the terraform files.

example - i am creating resource group name "rg" in my main.tf file. i have to deploy it to test , prod and stage environment . i want to add some suffix to this name like "rg-test" ," rg-prod" in github action level.  
is it possible to create a environment variable in github action level and how it will access the terraform file(rg name). 

this is my main.tf file.
resource "azurerm_resource_group" "example" {
  name     = "rg"
  location = "West Europe"
 
}

Can we set anything on github action level so that whenever github action job for test env runs ,it will add ENV variable in the "name".
example - "rg-test" based on env name.

**Comment by nehabam** [comment]
this is github action file sample.

name: terraform

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the "main" branch
  push:
    branches: [ "test-git" ]
  pull_request:
    branches: [ "test-git" ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

env:
      ROOT_PATH: '${{github.workspace}}/terraform'
      ARM_CLIENT_ID: ${{secrets.AZURE_CLIENT_ID}}
      ARM_CLIENT_SECRET: ${{secrets.AZURE_CLIENT_SECRET}}
      ARM_SUBSCRIPTION_ID: ${{secrets.AZURE_SUBSCRIPTION_ID}}
      ARM_TENANT_ID: ${{secrets.AZURE_TENANT_ID}}
      env: 'test'

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  terraform_plan:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest
    

    defaults:
      run:
        shell: bash

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
    - name: Checkout
      uses: actions/checkout@v2
    - name: Setup terraform
      uses: hashicorp/setup-terraform@v1
      with:
        cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
    - name: terraform init
      run: terraform init
      working-directory: ${{ env.ROOT_PATH }}
    - name: terraform Plan
      run: terraform plan  -out=plan.tfplan
      working-directory: ${{ env.ROOT_PATH }}

**Comment by Jamonygr** [comment]
Yes. Make the suffix a Terraform variable, then pass it from GitHub Actions with `TF_VAR_environment`.

```hcl
variable "environment" {
  type = string
}

resource "azurerm_resource_group" "example" {
  name     = "rg-${var.environment}"
  location = "West Europe"
}
```

```yaml
env:
  TF_VAR_environment: test
```

For prod/stage, set `TF_VAR_environment` to a different value at the job or environment level. Terraform will read it automatically during `plan` and `apply`.
